OsmAnd-java/src/main/java/net/osmand/CollatorStringMatcher.java

```diff
@@ -69,7 +69,7 @@ public static boolean cmatches(Collator collator, String fullName, String part,
 		}
 		switch (mode) {
 		case CHECK_CONTAINS:
-			return ccontains(collator, fullName, part); 
+			return ccontains(collator, fullName, part) || ccontains(collator, part, fullName);
```

Comment:
this change is big will require a deep understanding consequencies if it's really needed